src/cautiousrobot/__main__.py

```diff
@@ -88,99 +88,107 @@ def download_images(data, img_dir, log_filepath, error_log_filepath, filename =
+                                            )
+                        update_log(log = log_data, index = i, filepath = log_filepath)
+                        
+                        #create the appropriate folders if necessary
+                        
+                        if os.path.exists(image_dir_path) != True:
+                            os.makedirs(image_dir_path, exist_ok=False)
+                        
+                        # save full size image to appropriate folder
+                        with open(f"{image_dir_path}/{image_name}", "wb") as out_file:
+                            shutil.copyfileobj(response.raw, out_file)
 
-                    else:
-                        time.sleep(wait)
-                else: #other fail, eg. 404
-                    redo = False
-                    log_errors = log_response(log_errors,
+                    # check for too many requests
+                    elif response.status_code in REDO_CODE_LIST:
+                        redo = True
+                        max_redos -= 1
+                        if max_redos <= 0:
+                            log_errors = log_response(log_errors,
                                             index = i,
                                             image = image_name,
-                                            url = url,
+                                            file_path= url,
                                             response_code = response.status_code)
-                    update_log(log = log_errors, index = i, filepath = error_log_filepath)
+                            update_log(log = log_errors, index = i, filepath = error_log_filepath)
+    
+                        else:
+                            time.sleep(wait)
+                    else: #other fail, eg. 404
+                        redo = False
+                        log_errors = log_response(log_errors,
+                                                index = i,
+                                                image = image_name,
+                                                file_path = url,
+                                                response_code = response.status_code)
+                        update_log(log = log_errors, index = i, filepath = error_log_filepath)
+    
+                    del response
+                
+            if downsample:
+                # Since we have image resize within a try and log failure, seems reasonable to not check for the image again.
+                downsample_dir_path = downsample_path
+                if subfolders:
+                    downsample_dir_path = downsample_path + "/" + data[subfolders][i]
+                if os.path.exists(downsample_dir_path + "/" + image_name):
+                    # Don't overwrite resized images either
+                    continue
+                
+                if not os.path.exists(downsample_dir_path):
+                    os.makedirs(downsample_dir_path, exist_ok=False)
```

Comment:
I think this should actually stay in the `downsample_and_save_image` function; do you mind moving it back?

Reply:
Hello Elizabeth,

I update the code to this 
   
# Check for img_dir
    img_dir = args.output_dir
    if os.path.exists(img_dir):
        sys.exit(f"'{img_dir}' already exists. Exited without executing.")

Reply:
I moved below code from main to downsample_and_save_image function. 

                if not os.path.exists(downsample_dir_path):
                    os.makedirs(downsample_dir_path, exist_ok=False)